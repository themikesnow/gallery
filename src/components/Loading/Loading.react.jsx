import React from 'react';

const Loading = (props) => {
  if (props.isIcon) {
    return (
      <i className="fa fa-refresh fa-spin fa-lg fa-fw" />
    );
  }

  return (
    <div className="loading">
      <i className="fa fa-refresh fa-spin fa-3x fa-fw" />
    </div>
  );
};

Loading.propTypes = {
  isIcon: React.PropTypes.bool,
};

Loading.defaultProps = {
  isIcon: false,
};

export default Loading;
