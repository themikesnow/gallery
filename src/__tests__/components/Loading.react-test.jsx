import React from 'react';
import renderer from 'react-test-renderer';
import Loading from '../../components/Loading/Loading.react';

describe('<Loading />', () => {
  it('Match Snapshot', () => {
    let component = renderer.create(<Loading />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(<Loading isIcon />);
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
