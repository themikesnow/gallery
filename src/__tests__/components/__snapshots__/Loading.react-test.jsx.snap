// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Match Snapshot 1`] = `
<div
  className="loading"
>
  <i
    className="fa fa-refresh fa-spin fa-3x fa-fw"
  />
</div>
`;

exports[`<Loading /> Match Snapshot 2`] = `
<i
  className="fa fa-refresh fa-spin fa-lg fa-fw"
/>
`;
